src/ci_workflow/ci_check_list.py

```diff
@@ -14,6 +15,11 @@
 class CiCheckList(ABC):
     def __init__(self, component: Any, target: CiTarget) -> None:
         self.component = component
+        try:
+            self.component_ref_is_commit = True if len(self.component.ref) == 40 and int(self.component.ref, 16) else False
```

Comment:
I think the length check and hexa-decimal int check don't mean the ref exists in the repo. Maybe we say this is a valid sha1 commit here, but we would check if this commit ref exists in the component repo with the `check()` method below.

Reply:
The check starts on line 36 of src/ci_workflow/ci_check_list_source_ref.py.
The check here is to ensure it is sha1 before it reach the check stage.

Reply:
Talked to @zelinh we will change component_ref_is_commit to component_ref_is_sha1 to reduce confusion before reaching the actual check stage.